src/dialects/abstract/query-generator.js

```diff
@@ -2221,7 +2221,21 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
   selectFromTableFragment(options, model, attributes, tables, mainTableAs) {
     this._throwOnEmptyAttributes(attributes, { modelName: model && model.name, as: mainTableAs });
 
-    let fragment = `SELECT ${attributes.join(', ')} FROM ${tables}`;
+    let fragment = `SELECT`;
+
+    if (typeof options.maxExecutionTimeMs === 'number' && this.dialect.supports.maxExecutionTimeHint) {
+      const MINIMUM_EXECUTION_TIME_VALUE = 0;
+      const MAXIMUM_EXECUTION_TIME_VALUE = 4_294_967_295;
+
+      if (options.maxExecutionTimeMs >= MINIMUM_EXECUTION_TIME_VALUE
```

Comment:
Let's throw if someone uses a number with decimals, NaN, or the Infinities

```suggestion
      if (Number.isSafeInteger(options.maxExecutionTimeMs) && options.maxExecutionTimeMs >= MINIMUM_EXECUTION_TIME_VALUE
```